### PR TITLE
misc: use top-level await

### LINKED ISSUE #13964: use top-level await
1) change base tsconfig module to `es2020`
2) change all the things to use top level await

### PR DESCRIPTION
**Pull request by connorjclark**
(#13967 should land first)

Fixes #13964

I likely missed a few places where we could use top-level await.

FYI:
image

I verified that a node process exits with 1 and logs the error when a promise rejects, so no need to catch them.

### CODE REVIEW

File: lighthouse-core/scripts/lantern/collect/collect.js [hunk 1]
```diff
@@ -26,8 +26,7 @@ if (!process.env.WPT_KEY) throw new Error('missing WPT_KEY');
 const WPT_KEY = process.env.WPT_KEY;
 const DEBUG = process.env.DEBUG;
 
-/** @type {typeof common.ProgressLogger['prototype']} */
-let log;
+const log = new common.ProgressLogger();
```
Comment (connorjclark) [inline lighthouse-core/scripts/lantern/collect/collect.js:29]: drive by

**Comment by brendankenny** [comment]
> I verified that a node process exits with 1 and logs the error when a promise rejects, so no need to catch them.

[happens in Node 15](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V15.md#throw-on-unhandled-rejections---33021), sry :/

**Comment by connorjclark** [comment]
hmm, was it perhaps backported?

image

**Comment by connorjclark** [comment]
anyway, we could add `NODE_OPTIONS=--unhandled-rejections=` in CI for node 14.

**Comment by connorjclark** [comment]
checked our min node version 14.15 and also OK

image

@brendankenny I think that top-level awaited promises are not unconsidered "unhandled" on rejections.

**Pull request merged**